src/components/AllSpecies/AllSpecies.tsx

```diff
@@ -1,41 +1,101 @@
 import React from 'react'
-import { Heading, Select, Box } from '@chakra-ui/react'
+import { Heading, Select, Box, SimpleGrid } from '@chakra-ui/react'
 import { ModifiedCharacter } from '../../api/starWars-types'
-import useModifyCharacters, { SORT_FACTOR, SortFactor } from '../../hooks/useModifyCharacters'
+import useModifyCharacters, {
+  SORT_FACTOR,
+  FILTER_FACTOR,
+  SortFactor,
+  FilterFactor,
+} from '../../hooks/useModifyCharacters'
 import CharacterList from '../CharacterList/CharacterList'
+import Selector from '../UI/Selector/Selector'
 
 type AllSpeciesProps = {
   characters: ModifiedCharacter[]
 }
 
+const filterTargets = (characters: ModifiedCharacter[], filterFactor: FilterFactor) => {
+  const filterTargetsObject = characters.reduce(
```

Comment:
[dumb question] - why did you decide to go for `reduce()` instead of `filter()`? 👀

Reply:
That's simply because I can't come up with other solution...
All I want to do is just pick all filter targets without duplications.
I think this also works.
```typescript
const filterTargets = (characters: ModifiedCharacter[], filterFactor: FilterFactor) => {
const filterCharacters = characters.map((character)=>character[filterFactor])

// delete duplication
const setCharacters = new Set (filterCharacters)

// Convert to array
const arrayCharacters = Array.form(setCharacters)

return arrayCharacters
}
```